kernel/src/process_standard.rs

```diff
@@ -1739,8 +1750,12 @@ impl<C: 'static + Chip> ProcessStandard<'_, C> {
         let flash_start = process.flash.as_ptr();
         let app_start =
             flash_start.wrapping_add(process.header.get_app_start_offset() as usize) as usize;
-        let init_fn =
+        let init_addr =
             flash_start.wrapping_add(process.header.get_init_function_offset() as usize) as usize;
+        let fn_base = flash_start as usize;
+        let fn_len = process.flash.len();
+        let init_fn =
+            CapabilityPtr::new_with_metadata(init_addr as *const (), fn_base, fn_len, Execute);
```

Comment:
Have added a comment.

The code for in-memory processes is very similar (but the range is where the text has been copied to, rather than its source. Similar to how the init_addr would be in the copy, not the original, if there were no range indications here).